samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -167,6 +172,19 @@ public String getProcessorId() {
     return processorId;
   }
 
+  @Override
+  public Set<String> getProcessorNames() {
+    final Set<String> processorNames = new HashSet<>();
+    // Leader election could have happened by this time.
```

Comment:
could? Or should?
And it would be nice to fail here if leader election has not been done. This would require some additional work in ZkJobCoordinator. Please open another JIRA to track it.

Reply:
Opened jira to check if leader election is done and added it to comments.